### PR TITLE
Update for removed check for privilege escalation for creating a user and editing assigned roles

### PR DESCRIPTION
**Pull request by mrnyax**
Update API documentation for removed privilege escalation. 

With the changes, only Super Admin role is checked/required when creating or editing a user to assign the Super Admin role

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -3670,11 +3684,11 @@ For Core Platform versions 2021.05 to 2022.02, the following apply:
 
 + Response 403 (application/json)
 
-        // when you can't assign privileges you don't have when you edit users
+        // when you can't assign Super Administrator role because you don't have the Super Administrator role when you edit users
         {
         "links": {},
             "error": {
-                "detail": "It's not possible to make these changes. You can only create or edit a user with privileges that also apply to your user account.",
+                "detail": "It's not possible to make these changes. You need Super Administrator role to create or edit a user with this role.",
```
Comment (tkrisztian95) [inline apiary.apib:3691]: I'm not sure why this is here, but the request above (in line ~3547) is `Update password attribute`. I think this is not a possible response. I would suggest removing this response block from here.

**Review by tkrisztian95 — CHANGES_REQUESTED** [review]

File: apiary.apib [hunk 2]
```diff
@@ -3137,12 +3137,26 @@ For Core Platform versions 2021.05 to 2022.02, the following apply:
...
+        }
+
++ Response 403 (application/json)
+
+        // when external authentication is enabled, you shouldn't be able to change password through the API
+        {
+            "links": {},
+                "error": {
+                    "reference": "82a781cb-059a-4039-a17d-af8e20bec019",
+                    "detail": "The user isn't allowed to change the password.",
```
[the remark this answers is not in the document]
Comment (hjohannsen) [reply apiary.apib]: Fixed.

[the remark this answers is not in the document]
Comment (hjohannsen) [reply apiary.apib]: I guess @tkrisztian95 should resolve.

**Review by hjohannsen — APPROVED** [review]
Fine with me!

**Review by tkrisztian95 — APPROVED** [review]
Good.

**Pull request merged**